### ISSUE TITLE
Saving database fails intermittently due to ERROR_NOT_SUPPORTED

### ISSUE THREAD

**Issue by cedws** [x64dbg/x64dbg#2709]
Debugger version:
`July 1 2021`

Operating system version:
`Windows 10 Pro N x64`

Description of issue:
When stopping the debugger whilst attached a process, the database intermittently fails to save and shows this error:

![image](https://user-images.githubusercontent.com/38229097/129080982-b3084e49-38b9-46c4-9259-7da05ce70456.png)

Unfortunately this means any changes are lost. In addition, the database is sometimes being moved to a `.bak` file which means I need to click `Restore backup database` to get it back. Once I do this it's usually fine for a while.

My storage is fine and isn't failing or anything like that.

Logs:
```
Loading database from C:\[...omitted] 9812ms
...
Process stopped with exit code 0x0
Saving database to C:\[...omitted]
Failed to write database file !(GetLastError() = ERROR_NOT_SUPPORTED: The request is not supported.)
Debugging stopped!
```

Steps to reproduce:
* Have database compression enabled
* Have a large database (mine is 243MB, that's large right?)
* Attach to a process
* Mess around a bit
* Detach by hitting the Close button

I'm not sure if this will be reliably reproducible.

**Comment by cedws** [comment]
FWIW I reinstalled the program, reimported my database, and now it's shrunk to 31MB and I haven't encountered this issue again. Will post any more updates or findings.

**Comment by cedws** [comment]
It's happening again unfortunately.

**Comment by mrfearless** [comment]
Is x64dbg installed in any protected folders, like Program Files, ProgramData, Program Files (x86), Documents etc? 
Might also be worth adding an exclusion to Windows Defender or any anti virus programs to see if that helps.

**Comment by cedws** [comment]
Yes. In the past I installed this via Scoop which stores packages in its own directory in the user's home. It symlinks some paths that should be persisted so I switched to manual install just in case that was the cause of this issue. Now I have the archive extracted in Documents, but it is still occuring.

**Comment by mrfearless** [comment]
Could be that the x64dbg archive is still protected (file downloaded from internet), might have to right click and unprotect it before extraction. Only other thing to try is creating a manual x64dbg directory like c:\x64dbg and extracting x64dbg archive contents into that, then right click on folder make sure you have full control rights and perhaps full ownership of this folder and sub-folders. And barring any anti-virus product hooking that is preventing writing the database file, maybe some other program or product is hooking something as well? Thats all the suggestions I can think of right now.

**Comment by cedws** [comment]
Closed by mistake.

I'll do some more investigation and let you know.

**Comment by mrexodia** [comment]
I did some investigation and this happens when `CreateFile` or `WriteFile` fails with `GetLastError() == ERROR_NOT_SUPPORTED`. The relevant code is at https://github.com/x64dbg/x64dbg/blob/development/src/dbg/filemap.h#L59-L107 and as you can probably see there is nothing special going on there.

My current theory is that a buggy filter driver is causing this for some reason (or your HDD is dead and fails to create the file?). Could you share a full list of loaded drivers? I didn't try to reproduce this yet, but I can probably adjust the logs to give a little more information on the API that failed but I never saw this error when working with x64dbg.

**Comment by cedws** [comment]
Here's a list of drivers outputted from `driverquery`: https://gist.github.com/cedws/77a7b9581d4677ad420e4dce26cfa104
Nothing special in there that I can see.

I don't have any reason to suspect a storage medium problem, it's a new SSD and I haven't had problems with anything else.

The permissions on the `~/Documents/x64dbg` folder look normal.

**Comment by jimmyasmwest** [comment]
got this problem too guys. The database is failing to save everytime with "Failed to write database file !(GetLastError() = ERROR_NOT_SUPPORTED: The request is not supported.)"

This happened after x32dbg stopped responding after closing the exe.
After restoring the database, it never saves. Comments and bookmarks appear to be there still.
[exe.zip](https://github.com/x64dbg/x64dbg/files/11073670/exe.zip)
